### PR TITLE
PMM-9169: Add primitive binary type

### PR DESCRIPTION
**Pull request by oter**
Ticket: https://jira.percona.com/browse/PMM-9169

pmm-agent PR: https://github.com/percona/pmm-agent/pull/304
pmm-managed PR: https://github.com/percona/pmm-managed/pull/868

FB: https://github.com/Percona-Lab/pmm-submodules/pull/2149

### CODE REVIEW

**Review by BupycHuk — DISMISSED** [review]

**Review by BupycHuk — DISMISSED** [review]

File: api/agentpb/agent.proto [hunk 1]
```diff
@@ -126,6 +127,12 @@ message QueryActionMap {
   map<string, QueryActionValue> map = 1;
 }
 
+// QueryActionBinary represents primitive.Binary used in query Actions.
+message QueryActionBinary {
+  int32 subtype = 1;
```
Comment (artemgavrilov) [inline api/agentpb/agent.proto:132]: What is the `subtype` ?

**Comment by artemgavrilov** [comment]
Also why we got this problem only now? MonogDB added this binary data to response only in recent releases?

File: api/agentpb/agent.proto (hunk 1 above)
Comment (oter) [reply api/agentpb/agent.proto:132]: please see https://bsonspec.org/spec.html 
`Generic binary subtype`

**Comment by oter** [comment]
> Also why we got this problem only now? MonogDB added this binary data to response only in recent releases?

this issue is quite old :)
https://jira.percona.com/browse/SAAS-427

File: api/agentpb/query.go [hunk 2]
```diff
@@ -243,7 +255,15 @@ func makeInterface(value *QueryActionValue) (interface{}, error) {
...
-
+	case *QueryActionValue_Binary:
+		marshaled, err := json.Marshal(BinaryActionValue{
+			Subtype: int(v.Binary.Subtype),
+			Bytes:   v.Binary.Bytes,
+		})
+		if err != nil {
+			return nil, err
+		}
+		return marshaled, nil
```
Comment (artemgavrilov) [inline api/agentpb/query.go:266]: Minor: 
```suggestion
	case *QueryActionValue_Binary:
	    return  json.Marshal(BinaryActionValue{
			Subtype: int(v.Binary.Subtype),
			Bytes:   v.Binary.Bytes,
		})

```

**Review by artemgavrilov — DISMISSED** [review]

**Review by artemgavrilov — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**